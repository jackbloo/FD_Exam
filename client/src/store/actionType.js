export const LOGIN = 'LOGIN'
export const FETCHPASSWORD = 'FETCHPASSWORD'
export const LOGOUT = 'LOGOUT'
export const FETCHDETAIL = 'FETCH DETAIL'
export const SETEDIT = 'SETEDIT'
export const SHOW = 'SHOW'
export const UNSHOW = 'UNSHOW'
export const LOAD = 'LOAD'
export const UNLOAD = 'UNLOAD'
export const DELETE = 'DELETE'