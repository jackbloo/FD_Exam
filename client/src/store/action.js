import {LOGIN, LOGOUT, FETCHPASSWORD,SHOW,UNSHOW,LOAD,UNLOAD, FETCHDETAIL} from '../store/actionType'


export const show = () => ({type:SHOW})
export const unshow = () => ({type:UNSHOW})
export const login = () => ({type:LOGIN})
export const logout = () => ({type:LOGOUT})
export const load = () => ({type:LOAD})
export const unload = () => ({type: UNLOAD})
export const fetchPassword = (data) =>  ({type:FETCHPASSWORD, data})


// export const deleteOnePass = (id) => dispatch => {
//     dispatch({type:LOAD})
//     pm
//     .doc(id)
//     .delete()
//     .then(function() {
//         dispatch({type:UNLOAD})
//         toast.success('Your Password is deleted')
//     }).catch(function(error) {
//         toast.error(error)
//     });
// }
