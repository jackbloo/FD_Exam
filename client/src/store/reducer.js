import {GETDATA} from './actionType'

const initialState = {
    editorChoice:[],
    latestArticles:[],
    latestReview:[],
    navbar: [{id:'skincare'},{id:'make up'},{id:'body'},{id:'hair'},{id:'fragrance'},{id:'nails'},{id:'tools'},{id:'brands'}],
    linkData:[{id: 'About Us'}, {id: 'Terms & Conditions'}, {id: 'Awards'},{id: 'Feedback'},{id: 'Privacy Policy'}, {id: 'Newsletter'}, {id: 'Content'},{id: 'Help'}]
}

const reducer = (state = initialState, action) => {
    switch(action.type){
        case GETDATA:
            const data = action.data
            return {
                ...state,
                editorChoice: [...data["editor's choice"]],
                latestArticles:[...data["latest articles"]],
                latestReview:[...data["latest review"]]
            }
        default:
            return state     
    }
 
}

export default reducer