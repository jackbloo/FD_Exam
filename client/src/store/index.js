import { configureStore } from '@reduxjs/toolkit'
import reducer from '../store/reducer'

const store = configureStore({reducer: reducer})

export default store