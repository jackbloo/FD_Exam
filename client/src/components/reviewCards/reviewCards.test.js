import React from 'react';
import {render, screen} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import ReviewCard from './reviewCards'
import { Provider} from 'react-redux'
import store from '../../store'


describe('Unit Testing of ReviewCard', () => {
    const data = {
        user: 'Test-1',
        published_at: '2 hours ago',
        role:'Director',
        product:{
            rating: 4.3,
            name:'Test',
            description: 'test-1'
        },
        profile:['','','', '19-27'],
        comment:'Test-2w'
    }

    it('Should Render Card match snapshot', ()=>{

     render(
            <Provider store={store}>
                    <ReviewCard data={data}/>
            </Provider>
        )
        const card = screen.getByTestId('review-card')
        expect(card).toMatchSnapshot()
    }) 
    
    it('Should Render Card without crashing', ()=>{
     render(
            <Provider store={store}>
                    <ReviewCard data={data}/>
            </Provider>
        )
        const card = screen.getByTestId('review-card')
        expect(card).toBeInTheDocument()
        expect(card).toBeVisible()
        expect(card).toHaveTextContent("TestTest-2wRead MoreTest-1Combination skin, 19-27")
        expect(card).not.toHaveTextContent('Facebook.com')
    })

    
    it('Should Render review image', ()=>{
     render(
            <Provider store={store}>
                    <ReviewCard data={data}/>
            </Provider>
        )
        const card = screen.getByTestId('review-image')
        expect(card).toBeInTheDocument()
        expect(card).toBeVisible()
    })

    it('Should Render review user', ()=>{
        render(
               <Provider store={store}>
                       <ReviewCard data={data}/>
               </Provider>
           )
           const card = screen.getByTestId('review-user')
           expect(card).toBeInTheDocument()
           expect(card).toBeVisible()
       })

    it('Should Render review comment', ()=>{
     render(
            <Provider store={store}>
                    <ReviewCard data={data}/>
            </Provider>
        )
        const card = screen.getByTestId('review-comments')
        expect(card).toBeInTheDocument()
        expect(card).toBeVisible()
    })

    it('Should Render review description for product', ()=>{
     render(
            <Provider store={store}>
                    <ReviewCard data={data}/>
            </Provider>
        )
        const card = screen.getByTestId('review-description-product')
        expect(card).toBeInTheDocument()
        expect(card).toBeVisible()
    })
    
    it('Should Render review description', ()=>{
     render(
            <Provider store={store}>
                    <ReviewCard data={data}/>
            </Provider>
        )
        const card = screen.getByTestId('review-description')
        expect(card).toBeInTheDocument()
        expect(card).toBeVisible()
    })

    it('Should Render review name', ()=>{
        render(
               <Provider store={store}>
                       <ReviewCard data={data}/>
               </Provider>
           )
           const card = screen.getByTestId('review-name')
           expect(card).toBeInTheDocument()
           expect(card).toBeVisible()
       })


})

