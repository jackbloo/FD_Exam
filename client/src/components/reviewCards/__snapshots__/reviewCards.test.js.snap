// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Testing of ReviewCard Should Render Card match snapshot 1`] = `
<div
  class="sc-bdvvtL cUcYXm"
  data-testid="review-card"
>
  <div
    class="sc-dkPtRN emJlw"
  >
    <div
      class="sc-ksdxgE AsKOh"
    >
      <img
        class="sc-furwcr flMcvf"
        data-testid="review-image"
      />
      <div
        class="sc-crHmcD crpHvu"
      >
        <div
          class="sc-egiyK cksTTo"
          data-testid="review-name"
        >
          Test
        </div>
        <div
          class="sc-bqiRlB gAGPzQ"
          data-testid="review-description-product"
        />
      </div>
    </div>
    <div
      class="sc-pVTFL fnsqDJ"
    >
      <div
        class="sc-jrQzAO fFXmdK"
      >
        <div
          class="sc-iqseJM jBbDTV"
        >
          <i
            class="sc-kDTinF cFTyHc fa fa-star icon"
          />
          <i
            class="sc-kDTinF cFTyHc fa fa-star icon"
          />
          <i
            class="sc-kDTinF cFTyHc fa fa-star icon"
          />
          <i
            class="sc-kDTinF cFTyHc fa fa-star icon"
          />
          <i
            class="sc-kDTinF cFTyHc fa fa-star icon"
          />
        </div>
      </div>
    </div>
    <div
      class="sc-hBUSln jXsEcT"
      data-testid="review-comments"
    >
      Test-2w
    </div>
    <div
      class="sc-fotOHu hbRtTa"
    >
      Read More
    </div>
  </div>
  <div
    class="sc-gsDKAQ fzOFFr"
  >
    <div
      class="sc-gKclnd gMfVCB"
    >
      <img
        class="sc-iCfMLu jmApPc"
        src="/assets/NaN.png"
      />
    </div>
    <div
      class="sc-jRQBWg cvyqwW"
    >
      <div
        class="sc-hKwDye farejx"
        data-testid="review-user"
      >
        Test-1
      </div>
      <div
        class="sc-eCImPb dbRBnC"
        data-testid="review-description"
      >
        Combination skin, 19-27
      </div>
    </div>
  </div>
</div>
`;
